### DISCUSSION TITLE
is there a way to publish the lambda with different aws-lambda-tools-xxx.json for different environment (prod, staging, dev)?

### DISCUSSION THREAD

**Discussion by xiaohui-hiwintech** [aws/aws-lambda-dotnet#1176] [Q&A] [answered]
We have three environments (prod, staging, dev), is there a way that we can use different `aws-lambda-tools-xxx.json` to publish to different environments in VS2022? If not, is there any other way we can achieve this goal? thank you so much.

**Comment by ashishdhingra** [comment, marked as answer]
@xiaohui-gsv I do not think you could use different `aws-lambda-tools.default.json` per environment? However, the AWS Extensions for .NET CLI tooling supports `--config-file` command line switch to override the default config file (refer https://github.com/aws/aws-extensions-for-dotnet-cli/issues/54#issuecomment-456219605).

I came across the link https://github.com/aws/aws-lambda-dotnet/issues/126#issuecomment-305564737 that might be helpful.

**Reply by xiaohui-hiwintech** [reply]
Cool, thank you so much, it works by using the `dotnet lambda deploy-function --config-file aws-lambda-tools.xxxx.json`, thank you so much.
